### PR TITLE
Closes #779: Migration for D7 column image paragraph type (to AZ Text).

### LINKED ISSUE #779: Migrate Column Image paragraph type
See migration mapping doc for details: https://docs.google.com/spreadsheets/d/134UMiYtE8f1YAKYgSukL2JxFzDJF6-TgCAvYFC-vxqQ/edit#gid=0

### PR DESCRIPTION
**Pull request by smitahantzios**
Migration for Paragraph Column Image to AZ Text

## Description
Migration for Paragraph Column Image to AZ Text

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // Merging the data into one field.
+    $value['value'] = '';
+    // Getting the media into text.
+    if (isset($value[0])) {
+      $media = Media::load($value[0]);
+      if (!empty($media)) {
+        $value['value'] .= '<drupal-media data-align="center" data-entity-type="media" data-entity-uuid="' . $media->get('uuid')->value . '" data-view-mode="az_large"></drupal-media>';
```
Comment (danahertzberg) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php:29]: `data-align` should be none (or left if none isn't an option)

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_column_image.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        plugin: migration_lookup
+        source: fid
+        migration: 
+          - az_media
+
+  field_az_text_area: 
+    plugin: paragraphs_column_image_field_merge
+    source: '@temp_photo'
+    caption: field_uaqs_caption_text
+    credit: field_uaqs_image_credit
```
Comment (danahertzberg) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_column_image.yml:32]: Just verifying this is _not_ modifying the caption or credit fields on the media entity, correct?

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php (hunk 1 above)
Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php:29]: @danahertzberg  Will add that in..thanks

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_column_image.yml (hunk 2 above)
Comment (joeparsons) [reply modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_column_image.yml:32]: > Just verifying this is _not_ modifying the caption or credit fields on the media entity, correct?

Looks that way to me.

Comment (smitahantzios) [reply modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_column_image.yml:32]: Yes, it is not. We are appending the values as in the mapping excel sheet.

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php (hunk 1 above)
Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsColumnImageFieldMerge.php]: Done

**Review by danahertzberg — APPROVED** [review]
Looks good from a requirements perspective.

**Review by tadean — APPROVED** [review]

**Pull request merged**